pkg/rpcclient/notary/actor.go

```diff
@@ -0,0 +1,315 @@
+	// transaction by an appropriate actor. Irrespective of this setting
+	// (which defaults to nil) NotaryAssisted, NotValidBefore and Conflicts
+	// attributes are always added.
+	FbAttributes []transaction.Attribute
+	// FbScript is the script to use in the Notarize convenience method, it
+	// defaults to a simple RET instruction (doing nothing).
+	FbScript []byte
+	// FbSigner is the second signer to be used for the fallback transaction.
+	// By default it's derived from the account and has None scope, it has
+	// to be a simple signature account, but this setting allows you to give
+	// it some other scope to be used in complex fallback scripts.
+	FbSigner actor.SignerAccount
+	// MainAttribtues are additional attributes to be added into main
+	// transaction by an appropriate actor. Irrespective of this setting
+	// (which defaults to nil) NotaryAssisted attribute is always added.
+	MainAttributes []transaction.Attribute
+	// MainCheckerModifier will be used by the main Actor when creating
+	// transactions. It defaults to using [actor.DefaultCheckerModifier]
+	// for result check and adds MaxNotValidBeforeDelta to the
+	// ValidUntilBlock transaction's field. Only override it if you know
+	// what you're doing.
+	MainCheckerModifier actor.TransactionCheckerModifier
+	// MainModifier will be used by the main Actor when creating
+	// transactions. By default it adds MaxNotValidBeforeDelta to the
+	// ValidUntilBlock transaction's field. Only override it if you know
+	// what you're doing.
+	MainModifier actor.TransactionModifier
+}
+
+// RPCActor is a set of methods required from RPC client to create Actor.
+type RPCActor interface {
+	actor.RPCActor
+
+	SubmitP2PNotaryRequest(req *payload.P2PNotaryRequest) (util.Uint256, error)
+}
+
+// NewDefaultActorOptions returns the default Actor options. Internal functions
+// of it need some data from the contract, so it should be added.
+func NewDefaultActorOptions(reader *ContractReader, acc *wallet.Account) ActorOptions {
+	opts := ActorOptions{
+		FbScript: []byte{byte(opcode.RET)},
+		FbSigner: actor.SignerAccount{
+			Signer: transaction.Signer{
+				Account: acc.Contract.ScriptHash(),
+				Scopes:  transaction.None,
+			},
+			Account: acc,
+		},
+		MainModifier: func(t *transaction.Transaction) error {
+			nvbDelta, err := reader.GetMaxNotValidBeforeDelta()
+			if err != nil {
+				return fmt.Errorf("can't get MaxNVBDelta: %w", err)
+			}
+			t.ValidUntilBlock += nvbDelta
+			return nil
+		},
+	}
+	opts.MainCheckerModifier = func(r *result.Invoke, t *transaction.Transaction) error {
+		err := actor.DefaultCheckerModifier(r, t)
+		if err != nil {
+			return err
+		}
+		return opts.MainModifier(t)
+	}
+	return opts
+}
+
+// NewActor creates a new notary.Actor using the given RPC client, the set of
+// signers for main transactions and the simple (one plain signature) account
+// that will sign notary requests. The set of signers will be extended by the
```

Comment:
Just a question to ensure I understand the idea: to properly use this API I should construct the list of main signers  containing a combination of the following accounts: `FakeSimpleAccount`, `FakeContractAccount`, `FakeMultisigAccount` (to be added), simple account, multisig account, contract account. And at least one of these accounts must be not-a-fake, right?

Reply:
Not really, one regular multisig account (+a simple one) works too (it's also tested somewhere this way). The purpose of these fakes is to make `SignTx()` work for cases where we don't have any key.

Reply:
Actually, this situtation probably can be improved with interfaces (rather than `*wallet.Account` mangling), but I've decided to make something that works first.

Reply:
These ready-to-use dummy accounts is a nice solution, they are well-documented and can be applied easily.